rfc/ipni-w3c.md

```diff
@@ -0,0 +1,346 @@
+# W3 IPNI Indexing RFC
+
+## Authors
+
+- [gammazero], [olizilla]
+
+## Goals
+
+A user can use a CID that identifies data stored on w3c to lookup the location where the data can be retrieved and content claims describing that data.
+
+The data's location can change and content claims about the data can change, at anytime during the data's storage lifetime, and queried for the data CIDs will be able to get the most locations and claims.
+
+This RFC describes how IPNI indexing can be used to lookup data locations and content claims for any CID within a CAR stored in w3c.
+
+## Abstract
+
+A user stores data as a CAR file. This CAR contains a set of multihashes for individual pieces of data within the CAR. 
+
+The user creates a an external CAR index [CARx-index](https://github.com/web3-storage/RFC/pull/9/files) that contains tuples of (CID, offset, length) of any items they intend to make individually retrievable from the car.
+
+With an [inclusion claim](https://hackmd.io/@gozala/content-claims?utm_source=preview-mode&utm_medium=rec#Inclusion-Claims), the user who created the CAR asserts that CAR contains a given set of multihashes, transitively, via the CARx-index.
+
+After storing the CAR and CARx-index, w3s creates a [location claim](https://hackmd.io/@gozala/content-claims?utm_source=preview-mode&utm_medium=rec#Location-Claims) for the CAR file and the CARx-index. Each location claim maps an item's CID to a URL where the item can be retrieved.
```

Comment:
I think linked inclusion claims were not a very good idea as in they don’t provide enough information to be all that useful and tie us to CARs. I think multihash inclusion claims that follow are better take.

specifically I would suggest adopting variant of inclusion claims where we claim that some that (multi)hash A is a byte range of content addressed by (multi)hash B

This also makes equivalence  claims obsolete because covering full range allows you to say this sha256 is same as this blake3 hash.

I’m still not sure if raw cid is better than multihash however. It seems more flexible but it also becomes vague if you start using e.g unixfs cids

Reply:
> we claim that some that (multi)hash A is a byte range of content addressed by (multi)hash B

The intent of this is to record where to retrieve the whole CAR file from, or the CARx-index from. So, byte ranges should not apply here. This is just saying "you can get the CAR file from this URL, and the signer is asserting this to be true".

> not sure if raw cid is better than multihash

I would prefer to use multihashes for everything and only use CIDs as a form of user input for compatibility with other CID-based systems.